drivers/flash/nrf_qspi_nor.c

```diff
@@ -607,6 +608,16 @@ static int qspi_erase(const struct device *dev, uint32_t addr, uint32_t size)
 		if (res == NRFX_SUCCESS) {
 			addr += adj;
 			size -= adj;
+
+			/* Erasing flash pages takes a significant period of time and the
+			 * flash memory is unavailable to perform additional operations
+			 * until done.
+			 */
+			rc = qspi_wait_while_writing(dev, K_MSEC(10));
```

Comment:
The flash chip on the nrf52840dk has the following typical-maximum erase times: 58-240ms (4KiB), 0.8-3.5s (64KiB), 2-4m (chip erase) so would think making this a configurable Kconfig would make more sense (or optionally making it multiple Kconfig options e.g. wait X period based on erase type then check every Y ms)

Reply:
That sounds like a great idea for a future improvement. The values tend to vary a lot between different flash chips, so I look forward to whatever solution is proposed to do that properly.

Reply:
I think that proper timeout would be number of pages in erased region times the most pessimistic timeout for page.
Via Kconfig we can only provide default value, as, with multiple instances of devices, each device may have different values. The times should be also available via SFDP.

Reply:
The expected duration for the operation is different from the poll rate.
If the expected duration is 10 seconds, you don't want to only poll every 10 seconds because then if its not ready on the first call you have to wait another 10 seconds, even though it is likely to be ready shortly.
So in an ideal world you would have a long sleep until you think it should be ready, then a relatively fast polling period until it is done.
Setting that up properly is much harder than simply "poll every 10ms" however.

This PR fixes a real problem (returning before the flash can be used again), and is an intermediate solution between "hard lock the CPU for 2 minutes while erasing" and "magic code that knows exactly when the operation is complete and polls once it is done".

Nothing here needs to be reverted if someone wants to move closer to the ideal solution, but that won't be me.
Happy to change the 10ms default to something else if that is considered more reasonable.

Reply:
> I think that proper timeout would be number of pages in erased region times the most pessimistic timeout for page.

This is also not the best idea IMO as the differences between "typical" and "maximum" erase duration are annoyingly large. MX25R6435:
![image](https://github.com/user-attachments/assets/770dbd8f-de52-4b88-b9bc-c8c0399de4ce)

I would much rather start polling at 0.8 seconds than wait until 3.5 seconds (to handle some pathological case) to erase 64KB for example.